lib/python/qmk/cli/lint.py

```diff
@@ -46,6 +48,11 @@ def lint(cli):
         ok = False
         cli.log.error('Missing %s', readme_path)
 
+    # Check keyboard name is valid
+    if not re.fullmatch(r"[a-z0-9][a-z0-9_/]*", cli.config.lint.keyboard):
```

Comment:
This doesn't actually match our requirements.

We probably want to iterate over each path component and use a check function that looks something like this:

```
from string import ascii_lowercase, digits

first_letter_ok = ascii_lowercase + digits
rest_ok = first_letter_ok + '_'

def is_kb_name(path_component):
    if path_component[0] not in first_letter_ok:
        return False

    for letter in path_component[1]:
        if letter not in rest_ok:
            return False

    return True
```

Reply:
Nice catch! Does something like the recent update look okay now?